src/order/lattice.lean

```diff
@@ -505,20 +505,23 @@ variables [lattice α] {a b c d : α}
 
 lemma inf_le_sup : a ⊓ b ≤ a ⊔ b := inf_le_left.trans le_sup_left
 
-@[simp] lemma inf_lt_sup : a ⊓ b < a ⊔ b ↔ a ≠ b :=
-begin
-  split,
-  { rintro H rfl, simpa using H },
-  { refine λ Hne, lt_iff_le_and_ne.2 ⟨inf_le_sup, λ Heq, Hne _⟩,
-    refine le_antisymm _ _,
-    exacts [le_sup_left.trans (Heq.symm.trans_le inf_le_right),
-      le_sup_right.trans (Heq.symm.trans_le inf_le_left)] }
-end
-
 @[simp] lemma sup_le_inf : a ⊔ b ≤ a ⊓ b ↔ a = b :=
 ⟨λ h, le_antisymm (le_sup_left.trans $ h.trans inf_le_right)
   (le_sup_right.trans $ h.trans inf_le_left), by { rintro rfl, simp }⟩
 
+@[simp] lemma inf_eq_sup : a ⊓ b = a ⊔ b ↔ a = b := by rw [←inf_le_sup.ge_iff_eq, sup_le_inf]
+@[simp] lemma sup_eq_inf : a ⊔ b = a ⊓ b ↔ a = b := eq_comm.trans inf_eq_sup
+@[simp] lemma inf_lt_sup : a ⊓ b < a ⊔ b ↔ a ≠ b := by rw [inf_le_sup.lt_iff_ne, ne.def, inf_eq_sup]
+
+lemma inf_eq_and_sup_eq_iff : a ⊓ b = c ∧ a ⊔ b = c ↔ a = c ∧ b = c :=
```

Comment:
Do we have a lemma that says `a = c ∧ b = c ↔ a = b ∧ b = c`? I feel like that would help with this kind of thing

Reply:
I cannot find any, and it seems like we would need many variants. Do you think this PR needs it?

Reply:
I think there are only two interesting variants here, `eq_and_eq_left : a = b ∧ a = c ↔a = b ∧ b = c ` and `eq_and_eq_right : a = c ∧ b = c ↔ a = b ∧ b = c`.

Can you add them locally for now and see if they actually help?

Reply:
Just tried now. It doesn't really help at all.

Reply:
I feel like there is something missing here to do this, but I suppose it's better to learn what Aesop can do automatically instead of having us stumble into it.

Reply:
Oh sorry, I finally understood what you meant. You wanted me to rewrite
```
a ⊓ b = c ∧ a ⊔ b = c ↔ a = c ∧ b = c
a ⊓ b = a ⊔ b ∧ a ⊔ b = c ↔ a = c ∧ b = c
a ⊓ b = a ⊔ b ∧ a ⊔ b = c ↔ a = b ∧ b = c
a = b ∧ a ⊔ b = c ↔ a = b ∧ b = c
```
but I still don't really see how it would work nicely for the last step.